src/script/lua_api/l_util.cpp

```diff
@@ -245,6 +246,35 @@ int ModApiUtil::l_get_hit_params(lua_State *L)
 	return 1;
 }
 
+// check_password_entry(name, entry, password)
+int ModApiUtil::l_check_password_entry(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string name = luaL_checkstring(L, 1);
+	std::string entry = luaL_checkstring(L, 2);
+	std::string password = luaL_checkstring(L, 3);
+
+
+	if (base64_is_valid(entry)) {
+		std::string gen_entry = translate_password(name, password);
+		lua_pushboolean(L, gen_entry == entry);
+		return 1;
+	}
+
+	std::string bytes_s;
+	std::string bytes_v;
+
+	if (!decode_srp_verifier_and_salt(entry, &bytes_s, &bytes_v)) {
+		// invalid format, return nil
```

Comment:
Because the mod author might also write something like if ```minetest.check_password_entry(..) == false then
-- fail authentication here
else
...
```
Since this is security-critical code, it better be as screwup-proof as possible.